### PR TITLE
stm32g4: adc: Add DTS for ADC345

### PR DESCRIPTION
**Pull request by pdietl**
Add DTS information for ADCs 3, 4, and 5 for stm32g4

Signed-off-by: Pete Dietl <petedietl@gmail.com>

### CODE REVIEW

File: dts/arm/st/g4/stm32g4.dtsi [hunk 1]
```diff
@@ -93,6 +93,36 @@
...
 
+		adc3: adc@50000400 {
+			compatible = "st,stm32-adc";
+			reg = <0x50000400 0x100>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00004000>;
+			interrupts = <47 0>;
+			status = "disabled";
+			label = "ADC_3";
+			#io-channel-cells = <1>;
+		};
```
Comment (erwango) [inline dts/arm/st/g4/stm32g4.dtsi:104]: To be moved to g473, g474, g483, g484, g491, g4a1

File: dts/arm/st/g4/stm32g4.dtsi [hunk 2]
```diff
@@ -93,6 +93,36 @@
...
+
+		adc5: adc@50000600 {
+			compatible = "st,stm32-adc";
+			reg = <0x50000600 0x100>;
+			clocks = <&rcc STM32_CLOCK_BUS_AHB2 0x00004000>;
+			interrupts = <62 0>;
+			status = "disabled";
+			label = "ADC_5";
+			#io-channel-cells = <1>;
+		};
```
Comment (erwango) [inline dts/arm/st/g4/stm32g4.dtsi:124]: To be moved to g473, g474, g483, g484

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by FRASTM** [comment]
The ADC 3 4 5 share a common bank of 1KB at  0x50000400. In the Table 166. ADC global register map from 0x50000400
Offset 0x000 - 0x0FC  stores the Master ADC3 registers
Offset 0x100 - 0x1FC stores the Slave ADC4  registers 
Offset 0x200 - 0x2FC is Reserved/single ADC5 registers
Offset 0x300 - 0x30C stores Master and slave ADCs common registers
So that ADC 3  and 4  have some of their common registers at address 0x50000400 + 0x300,
Should  ADC 3 and 4 access that area ?

**Comment by pdietl** [comment]
@FRASTM It looks like 3, 4, & 5 have common registers at + 0x300. The same goes for ADC 1 & 2, and we don't indicate the offset of that shared memory region either. Is it possible to specify multiple `reg`s?

Also, we don't need to know that offset, because it is taken care of by the LL library: `LL_ADC_GetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc));`

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**